contracts/utils/sign.py

```diff
@@ -24,8 +24,30 @@ def eth_message_hex(msg: str) -> bytes:
     return sol_sha3(msg_hex)
 
 
-def sign(data: str, private_key_seed_ascii: str):
-    data = eth_message_hex(data)
+def eth_signed_typed_data_message(types, names, data) -> bytes:
+    """
+    types e.g. ('address', 'uint', ('uint', 32))
+    names e.g. ('receiver', 'block_created', 'balance')
+    data e.g. ('0x5601ea8445a5d96eeebf89a67c4199fbb7a43fbb', 3000, 1000)
+    """
+    assert len(types) == len(data) == len(names), 'Check argument length.'
```

Comment:
I think the assert message should not read like a comment, but like an error message. So something like:
`'Argument tuple length mismatch'`